File: app/src/main/java/com/example/odm/garbagesorthelper/model/source/local/LocalGarbageHistoryDataSource.java
package com.example.odm.garbagesorthelper.model.source.local;

import com.example.odm.garbagesorthelper.model.entity.GarbageSearchHistory;
import com.orhanobut.logger.Logger;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Single;

/**
 * @description: 垃圾搜索历史数据源
 * @author: ODM
 * @date: 2019/11/10
 */
public class LocalGarbageHistoryDataSource implements GarbageHistoryDataSource {

    private final GarbageHistoryDao mDao;

    public LocalGarbageHistoryDataSource(GarbageHistoryDao garbageHistoryDao) {
        this.mDao = garbageHistoryDao;
    }

    @Override
    public Flowable<List<GarbageSearchHistory>> getAllGarbageHistory() {
        return mDao.getAllGarbageHistory();
    }

    @Override
    public Completable deleteAllGarbageHistory() {
         return mDao.deleteAllGarbageHistory();
    }

    @Override
    public Completable insertGarbageHistory(GarbageSearchHistory... histories) {

        return mDao.insertGarbageHistory(histories);
    }


    @Override
    public Completable delete(GarbageSearchHistory... histories) {
        return mDao.delete(histories);
    }

    @Override
    public Single<GarbageSearchHistory> getGarbageHistoryByName(String name) {
        return mDao.getGarbageHistoryByName(name);
    }

    @Override
    public  Flowable<List<GarbageSearchHistory>> loadAllByType(int type) {
        return mDao.loadAllByType(type);
    }

    @Override
    public Flowable<GarbageSearchHistory> getGarbageHistoryById(int id) {
        return mDao.getGarbageHistoryById(id);
    }
}

File: app/src/main/java/com/example/odm/garbagesorthelper/model/source/local/LocalGarbageHistoryDataSource.kt
package com.example.odm.garbagesorthelper.model.source.local

import com.example.odm.garbagesorthelper.model.entity.GarbageSearchHistory
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Single

/**
 * @description: 垃圾搜索历史数据源
 * @author: ODM
 * @date: 2019/11/10
 */
class LocalGarbageHistoryDataSource(private val mDao: GarbageHistoryDao) : GarbageHistoryDataSource {
    override val allGarbageHistory: Flowable<List<GarbageSearchHistory?>?>?
        get() = mDao.allGarbageHistory

    override fun deleteAllGarbageHistory(): Completable? {
        return mDao.deleteAllGarbageHistory()
    }

    override fun insertGarbageHistory(vararg histories: GarbageSearchHistory?): Completable? {
        return mDao.insertGarbageHistory(*histories)
    }

    override fun delete(vararg histories: GarbageSearchHistory?): Completable? {
        return mDao.delete(*histories)
    }

    override fun getGarbageHistoryByName(name: String?): Single<GarbageSearchHistory?>? {
        return mDao.getGarbageHistoryByName(name)
    }

    override fun loadAllByType(type: Int): Flowable<List<GarbageSearchHistory?>?>? {
        return mDao.loadAllByType(type)
    }

    override fun getGarbageHistoryById(id: Int): Flowable<GarbageSearchHistory?>? {
        return mDao.getGarbageHistoryById(id)
    }

}